src/Cassandra/Connections/HostConnectionPool.cs

```diff
@@ -942,7 +1034,28 @@ private async Task<IConnection> GetConnectionFromHostAsync(
         /// </summary>
         public async Task Warmup()
         {
+            if (!_poolingOptions.GetDisableShardAwareness())
+            {
+                try
+                {
+                    var c = await CreateOpenConnection(false, false, false).ConfigureAwait(false);
+                    var _shardingInfo = c.ShardingInfo();
+                    if (_shardingInfo != null)
+                    {
+                        shardingInfo = _shardingInfo;
+                        _expectedConnectionLength = _shardingInfo.ScyllaNrShards;
+                    }
+                    c.Dispose();
```

Comment:
It is known that scylla has problems with handling reconnects, it would be better not to close this connection, but rather add it to the pool.
You can read shard from this connection and save it for the shard